build/api-docs-generator.js

```diff
@@ -0,0 +1,655 @@
+    console.error('Error generating or processing MDX files:', error);
+  }
+}
+
+/**
+ * Reads JSDoc configuration file.
+ * @returns {Object} JSDoc configuration object
+ */
+function readJsdocFile() {
+  if (fs.existsSync(jsdocFile)) {
+    return JSON.parse(fs.readFileSync(jsdocFile, 'utf-8'));
+  } else {
+    console.error("Jsdoc.json not found! Make sure it is inside the same directory as of the script!");
+  }
+}
+
+/**
+ * Reads the configuration file.
+ * @returns {Object} Configuration object
+ */
+function readConfigFile() {
+  if (fs.existsSync(configFile)) {
+    return JSON.parse(fs.readFileSync(configFile, 'utf-8'));
+  } else {
+    console.error("Config.json not found! Make sure it is inside the same directory as of the script!");
+  }
+}
+
+/**
+ * Copies JS files from source to destination directory.
+ * 
+ * @param {string} source - Source directory
+ * @param {string} destination - Destination directory
+ */
+function copyJsFiles(source, destination) {
+  const entries = fs.readdirSync(source, { withFileTypes: true });
+
+  entries.forEach(entry => {
+    const srcPath = path.join(source, entry.name);
+    const destPath = path.join(destination, entry.name);
+
+    if (entry.isDirectory()) {
+      if (!fs.existsSync(destPath)) {
+        fs.mkdirSync(destPath, { recursive: true });
+      }
+      copyJsFiles(srcPath, destPath);
+    } else if (path.extname(entry.name) === '.js') {
+      fs.copyFileSync(srcPath, destPath);
+    }
+  });
+}
+
+/**
+ * Recursively gets all JS files in a directory.
+ * 
+ * @param {string} dir - Directory to search
+ * @returns {string[]} Array of file paths
+ */
+function getJsFiles(dir) {
+  let results = [];
+  const entries = fs.readdirSync(dir, { withFileTypes: true });
+
+  for (const entry of entries) {
+    const fullPath = path.join(dir, entry.name);
+    if (entry.isDirectory()) {
+      results = results.concat(getJsFiles(fullPath));
+    } else if (path.extname(fullPath) === '.js') {
+      results.push(fullPath);
+    }
+  }
+
+  return results;
+}
+
+/**
+ * Processes a single JS file to generate MDX.
+ * It processes one JS file at a time, as one JS file may create multiple MDX files, so after creating, we need to merge all the MDX files created a single JS file.
+ * 
+ * @param {string} file - Path to the JS file
+ * @param {Object} config - Configuration object
+ */
+function processJsFile(file, config) {
+  const relativeDir = path.dirname(path.relative(devApiDir, file));
+  const fileName = path.basename(file, '.js');
+
+  // Copy JS file to temp directory
+  fs.copyFileSync(file, path.join(tempDir, path.basename(file)));
+
+  // Set unique outDir for this file
+  const outDir = path.join(mdxApiDir, relativeDir, fileName);
+  config.outDir = path.relative(__dirname, outDir);
+  fs.writeFileSync(configFile, JSON.stringify(config, null, 2));
+
+  // Run jsdoc-to-mdx
+  execSync(`npx jsdoc-to-mdx -c ${path.relative(__dirname, configFile)}`, { cwd: __dirname });
```

Comment:
We need to add the `jsdoc-to-mdx` deps in the package.json of phoenix code for this to  work right?

Reply:
Yes, jsdoc-to-mdx should be added as a dependency in the package.json file.

You can try running `npm install jsdoc-to-mdx`—it should automatically handle the setup. Once that's done, the script should run without any issues.